### PR TITLE
Fix & Test #613

### LINKED ISSUE #613: ParameterizedTypeName.get loses nullability information
```kotlin
ParameterizedTypeName.get(Any::class, true, emptyList())
```
 will return the `TypeName` for `Any`, not for `Any?`.

This is because the `nullable` parameter will be discarded if the `typeArguments` are empty in https://github.com/square/kotlinpoet/blob/master/src/main/java/com/squareup/kotlinpoet/TypeName.kt#L583 : 

```kotlin
  /** Returns a type name equivalent to type with given list of type arguments.  */
    internal fun get(
      type: KClass<*>,
      nullable: Boolean,
      typeArguments: List<KTypeProjection>
    ): TypeName {
      if (typeArguments.isEmpty()) {
        return type.asTypeName()
      }

     […]
```

### PR DESCRIPTION
**Pull request by jGleitz**
fixes #613

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/TypeName.kt [hunk 1]
```diff
@@ -580,7 +580,7 @@ class ParameterizedTypeName internal constructor(
       typeArguments: List<KTypeProjection>
     ): TypeName {
       if (typeArguments.isEmpty()) {
-        return type.asTypeName()
+        return type.asTypeName().run { if (nullable) copy(nullable = true) else this }
```
Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/TypeName.kt]: Sweet!

File: src/test/java/com/squareup/kotlinpoet/AbstractTypesTest.kt [hunk 2]
```diff
@@ -193,6 +194,9 @@ abstract class AbstractTypesTest {
         .isEqualTo("kotlin.collections.Map<kotlin.String, *>")
     assertThat(Map.Entry::class.createType(listOf(KTypeProjection(KVariance.INVARIANT, String::class.createType(emptyList())), KTypeProjection.STAR)).asTypeName().toString())
         .isEqualTo("kotlin.collections.Map.Entry<kotlin.String, *>")
+    assertThat(Any::class.starProjectedType.withNullability(true).asTypeName().toString())
+        .isEqualTo("kotlin.Any?")
+
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/AbstractTypesTest.kt]: nit: empty line

Comment (jGleitz) [reply src/test/java/com/squareup/kotlinpoet/AbstractTypesTest.kt]: removed it.

**Comment by jGleitz** [comment]
I have signed the CLA.

**Pull request merged**